frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/delete-block-pool-modal.tsx

```diff
@@ -0,0 +1,185 @@
+  const { blockPoolConfig, cancel, close, handlePromise, inProgress } = props;
+
+  const [state, dispatch] = React.useReducer(blockPoolReducer, blockPoolInitialState);
+  const [scNames, setScNames] = React.useState<string[]>([]);
+
+  const [cephClusters, isLoaded, loadError] = useK8sWatchResource<CephClusterKind[]>(
+    cephClusterResource,
+  );
+  const [scResources, scLoaded, scLoadError] = useK8sGet<ListKind<OcsStorageClassResource>>(
+    StorageClassModel,
+  );
+  const [pvcResources, pvcLoaded, pvcLoadError] = useK8sGet<ListKind<PersistentVolumeClaimKind>>(
+    PersistentVolumeClaimModel,
+  );
+  const cephCluster: CephClusterKind = useDeepCompareMemoize(cephClusters[0], true);
+
+  const populateBlockPoolData = React.useCallback(
+    (poolConfig: StoragePoolKind) => {
+      dispatch({
+        type: BlockPoolActionType.SET_POOL_NAME,
+        payload: poolConfig?.metadata.name,
+      });
+      dispatch({
+        type: BlockPoolActionType.SET_POOL_REPLICA_SIZE,
+        payload: poolConfig?.spec.replicated.size.toString(),
+      });
+      dispatch({
+        type: BlockPoolActionType.SET_POOL_COMPRESSED,
+        payload: poolConfig?.spec.compressionMode === COMPRESSION_ON,
+      });
+      // Already existing pool may not have any deviceClass, Default is SSD
+      poolConfig?.spec.deviceClass &&
+        dispatch({
+          type: BlockPoolActionType.SET_POOL_VOLUME_TYPE,
+          payload: poolConfig?.spec.deviceClass,
+        });
+    },
+    [dispatch],
+  );
+
+  React.useEffect(() => {
+    // restrict pool management for default pool and external cluster
+    cephCluster?.metadata.name === CEPH_EXTERNAL_CR_NAME ||
+    blockPoolConfig?.metadata.ownerReferences?.find(
+      (ownerReference) => ownerReference.name === OCS_INTERNAL_CR_NAME,
+    )
+      ? dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.NOTALLOWED })
+      : populateBlockPoolData(blockPoolConfig);
+  }, [blockPoolConfig, cephCluster, isLoaded, loadError, populateBlockPoolData]);
+
+  // find pvc using sc
+  const checkPvcStatus = React.useCallback(
+    (scNameList: string[]) => {
+      const pvcScNames = pvcResources?.items?.map((pvcConfig) => pvcConfig?.spec?.storageClassName);
+      // intersection of scNames and pvcScNames
+      const usedScNames = scNameList.filter((scName) => pvcScNames.includes(scName));
+      if (usedScNames.length) {
+        dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.BOUNDED });
+        setScNames(usedScNames);
+      }
+    },
+    [pvcResources],
+  );
+
+  // find storage class using pool
+  React.useEffect(() => {
+    if (scLoaded && pvcLoaded)
+      checkPvcStatus(
+        scResources?.items
+          ?.filter((scConfig) => state.poolName === scConfig?.parameters?.pool)
+          .map((scConfig) => scConfig?.metadata?.name) || [],
+      );
+  }, [scResources, scLoaded, pvcResources, pvcLoaded, state.poolName, checkPvcStatus]);
```

Comment:
You can find a lot of these helpers in `selectors/index.ts` . You can leverage those and you should always create pure functions is possible.
For instance -> `getPVStorageClass`  for 
`const pvcScNames = pvcResources?.items?.map((pvcConfig) => pvcConfig?.spec?.storageClassName);`
And set the state `setScNames` in useEffect